src/niri.rs

```diff
@@ -2408,7 +2408,18 @@ impl Niri {
 
         // Check if some layer-shell surface is on top.
         let layers = layer_map_for_output(output);
-        let layer_under = |layer| layers.layer_under(layer, pos_within_output).is_some();
+        let layer_under = |layer| {
+            layers
+                .layer_under(layer, pos_within_output)
+                .and_then(|layer| {
+                    let layer_pos_within_output = layers.layer_geometry(layer)?.loc.to_f64();
```

Comment:
This can remain an unwrap(). It is guaranteed by the fact that we found it in `layer_under()`.

Reply:
:+1: updated